File: add/codesnippet/VisualBasic/colors/sourcetoboolconverter.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections
Imports System.Globalization
Imports System.Reflection
Imports System.ComponentModel
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Data
Imports System.Windows.Media


Namespace SDKSample

	' One-way conversion between ColorItem.Source and bool -
	' used to disable editing for builtin colors

	Public Class SourceToBoolConverter
		Implements IValueConverter
		Private Function Convert(ByVal o As Object, ByVal type As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
			Dim source As ColorItem.Sources = CType(o, ColorItem.Sources)
			Return (source <> ColorItem.Sources.BuiltIn)
		End Function

		Private Function ConvertBack(ByVal o As Object, ByVal type As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
			Return Nothing
		End Function
	End Class
End Namespace

File: samples/snippets/csharp/VS_Snippets_Wpf/Colors/CSharp/SourceToBoolConverter.cs
using System;
using System.Collections;
using System.Globalization;
using System.Reflection;
using System.ComponentModel;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Data;
using System.Windows.Media;


namespace SDKSample
{

    // One-way conversion between ColorItem.Source and bool -
    // used to disable editing for builtin colors

    public class SourceToBoolConverter : IValueConverter
    {
        object IValueConverter.Convert(object o, Type type, object parameter, CultureInfo culture)
        {
            ColorItem.Sources source = (ColorItem.Sources)o;
            return (source != ColorItem.Sources.BuiltIn);
        }

        object IValueConverter.ConvertBack(object o, Type type, object parameter, CultureInfo culture)
        {
            return null;
        }
    }
}
